train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -43,7 +58,8 @@ private void enforceSpeedLimit() {
 
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;
+		refThread.start(); 	
```

Comment:
Start metódus a konstruktorban is meghívható, de így csak akkor indul a thread, ha változik a step